### ISSUE TITLE
When an order of a product's n quantity's is created, in shiprocket's order details we can see Unit price is calculated by the total price of  the product * Qty. Although the Net Total remains proper

### ISSUE THREAD

**Issue by RonakPanchal2** [Hemann55/medusa-fulfillment-shiprocket#2]
![Screenshot (97)](https://github.com/Hemann55/medusa-fulfillment-shiprocket/assets/139123252/e29dec31-f416-4b9b-95ce-ab3001367e23)

**Comment by rahatsayyed** [comment]
@RonakPanchal2 can you share your process of adding plugin I followed all the steps (env, config) but on admin page while adding the fulfilment option it says no options available
![Screenshot 2024-07-01 155241](https://github.com/Hemann55/medusa-fulfillment-shiprocket/assets/70068895/da20fa04-4201-4286-a51b-77df39306fc0)

**Comment by RonakPanchal2** [comment]
@youreachedrahat Check console what errors do you get. Sometimes too many requests are send to shiprocket api to get the available courier companies then it gives errors,  a rate limit may be set by shiprocket on their endpoints . Please check your console you will get an idea about the error. 
Also the plugin has some bugs in it as you proceed further you will get to know.  I have removed the plugin and added plugins files to my projects src file and fixed the bugs myself. You will need  to do the same till then keep developing till things go well as per design.

**Comment by RonakPanchal2** [comment]
@youreachedrahat Have you followed these steps https://www.npmjs.com/package/medusa-fulfillment-shiprocket 
After installation you need to add the plugins code in medusa-config.js file the code is mentioned in the npm docs. Every plugin you install you need to add it in the plugins array with the options the plugin requires. If you have done this part, please check your console for any errors

**Comment by rahatsayyed** [comment]
@RonakPanchal2 Yes I have added the plugin code in config as well as the Env variables.
in Region setting I am able to add the shiprocket but when adding Shipping Options, then it says no option available.

and after installing Shiprocket plugin and using it in particular region then for that region even payment options are gone
```
Fulfillment Method
no options

**Comment by RonakPanchal2** [comment]
@youreachedrahat  IDK where are you going wrong. The integration was quiet simple. Make sure you have added the shipocket credentials properly You will have to create a an API user in shiprocket and in the .env file use that api user's email and password along with channel_id. I think you are missing this as you are able to see plugin available in the fulllfillment providers mean the plugin is installed properly but the data could not be fetched from shiprocket. Due to wrong credenials

**Comment by rahatsayyed** [comment]
does `#` at the end of env variable can be a problem (I had wrong email in the env. I fixed it now it says this)
```
{ message: 'Wrong number of segments', status_code: 401 }
error:   Shiprocket Courier: Failed to retrieveAll
Error: Shiprocket Courier: Failed to retrieveAll
    at E:\Medusa\tradewave\node_modules\medusa-fulfillment-shiprocket\utils\shiprocket.js:308:25
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
::1 - - [03/Jul/2024:11:51:00 +0000] "GET /admin/regions/reg_01J1W6QN0KNP37JEG9BZ9Y8PWE/fulfillment-options HTTP/1.1" 500 86 "http://localhost:7001/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/126.0.0.0 Safari/537.36"

**Comment by RonakPanchal2** [comment]
Try directly making the request to the shiprocket apis using the credentials you are using in you env if it works as I can see 401 error means you are not authorized to make the req

**Comment by rahatsayyed** [comment]
hey thanks for the help it was `#` at the end of password it was ignored considering it as a comment I enclosed it in quotes `' .. # '` and works fine

One last question if you have already deployed can you tell me estimate it would cost for deployment and the monthly charges (I already have domain so you can skip it )
1. deployment on digitalOcean, AWS, Azure cost
2. the database cost
and if any montly charges
